collectfast/management/commands/collectstatic.py

```diff
@@ -32,13 +39,20 @@ def __init__(self, *args, **kwargs):
         self.tasks = []
         self.etags = {}
         self.collectfast_enabled = settings.enabled
-        if self.storage.preload_metadata is not True:
-            self.storage.preload_metadata = True
-            warnings.warn(
-                "Collectfast does not work properly without "
-                "`preload_metadata` set to `True` on the storage class. Try "
-                "setting `AWS_PRELOAD_METADATA` to `True`. Overriding "
-                "`storage.preload_metadata` and continuing.")
+        if settings.enabled:
+            if django_settings.STATICFILES_STORAGE not in VALID_BACKENDS:
+                raise RuntimeError(
+                    "Collectfast is intended to work with an S3 storage "
+                    "backend only."
+                )
```

Comment:
I don't think this is the desired behaviour. In case a storage backend is configured that isn't S3 we want to handle it gracefully. I've had a plan (in my head) for some time to introduce an abstraction layer for what strategy to use for different backends, but I think that's a much larger scope than what's needed here.

I believe all we want to do is skip the md5/etag comparison if the backend isn't s3. If it's feasible I think it'd be cool if we could still have threading enabled.

Reply:
Thanks for your comment @antonagestam.

Yes, I understand the desire to handle non-S3 backends gracefully. Just to verify I understood you correctly, you're saying that when `settings.enabled == True` we should handle things gracefully even if the storage backend isn't S3, right?
If that's the case, we must check the backend type before referencing `self.storage.preload_metadata`, otherwise we get an `AttributeError`. As a reminder, in https://github.com/antonagestam/collectfast/issues/120 you've suggested to do:

`if settings.enabled and self.storage.preload_metadata is not True:`

Instead of checking the storage type.

So, now I'm confused :-) If we want to allow `settings.enabled == True` + non-S3 backend, we must check the storage type, unless I'm missing something.

Could you please clarify the right approach, the way you see it? I'd love to solve this this way or the other, without increasing the scope of this bugfix too much.

Please let me know your thoughts.

Thanks!

Reply:
@antonagestam - are you suggesting something like:
````python
self.collectfast_enabled = settings.enabled
if not hasattr(self.storage, 'preload_metadata'):  # or S3 checks
    self.collectfast_enabled = False
    warnings.warn('Your current storage does not have preload_metadata!'
                  'Disabling collectfast and continuing...'
    )
if self.collectfast_enabled and self.storage.preload_metadata is not True:
    self.storage.preload_metadata = True
    warnings.warn(
        # ...
    )
````